app/controllers/homepage_controller.rb

```diff
@@ -61,13 +65,17 @@ def index
 
     search_result = find_listings(params, per_page, compact_filter_params, includes.to_set, location_in_use, keyword_in_use)
 
-    shape_name_map = all_shapes.map { |s| [s[:id], s[:name]]}.to_h
-
     if @view_type == 'map'
       viewport = viewport_geometry(params[:boundingbox], params[:lc], @current_community.location)
     end
 
-    if request.xhr? # checks if AJAX request
+    if FeatureFlagHelper.search_engine == :discovery
+      search_result.on_success { |listings|
+        render template: "search_page/search_page", locals: { bootstrapped_data: listings }
+      }.on_error {
+        render nothing: true, status: 500
+      }
+    elsif request.xhr? # checks if AJAX request
```

Comment:
New browse view paging is not designed yet, but it probably won't use this code path.